File: ide-features-trainer/src/org/jetbrains/training/commands/ShowLineNumberCommand.java
package org.jetbrains.training.commands;

/**
 * Created by karashevich on 30/01/15.
 */
public class ShowLineNumberCommand extends Command {

    public ShowLineNumberCommand(){
        super(CommandType.SHOWLINENUMBER);
    }

    @Override
    public void execute(ExecutionList executionList) {
        //Block caret and perform next command
//        ActionManager.getInstance().getAction()
        executionList.getEditor().getSettings().setLineNumbersShown(true);
        executionList.getElements().poll();
        startNextCommand(executionList);
    }
}

File: ide-features-trainer/src/training/commands/ShowLineNumberCommand.kt
/*
 * Copyright 2000-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
 */
package training.commands

import com.intellij.openapi.application.ApplicationManager
import com.intellij.openapi.editor.Editor

class ShowLineNumberCommand : Command(Command.CommandType.SHOWLINENUMBER) {

  override fun execute(executionList: ExecutionList) {
    //Block caret and perform next command
    //        ActionManager.getInstance().getAction()
    val editor = executionList.editor
    showLineNumber(editor)
    executionList.elements.poll()
    startNextCommand(executionList)
  }

  companion object {
    fun showLineNumber(editor: Editor) {
      ApplicationManager.getApplication().invokeAndWait {
        editor.settings.isLineNumbersShown = true
      }
    }
  }
}
